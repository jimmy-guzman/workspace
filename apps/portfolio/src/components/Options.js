import React from 'react'
import PropTypes from 'prop-types'
import { NavLink } from 'react-router-dom'
// import DaySVG from "./SVGs/DaySVG";
// import NightSVG from "./SVGs/NightSVG";

const Options = ({ isEnglish, updateLang }) => (
  <div className='options'>
    {/* <div className="options__day-night">
      <button className="btn__primary" aria-label="Day Mode">
        <DaySVG height="24" width="24" />
      </button>
      <button className="btn__primary" aria-label="Night Mode">
        <NightSVG height="24" width="24" />
      </button>
    </div> */}
    <NavLink exact to='/' className='home-link' activeClassName='active-home'>
      Jimmy Guzman
    </NavLink>
    <div className='options__lang'>
      <button
        className={!isEnglish ? 'btn__primary active' : 'btn__primary'}
        type='button'
        onClick={() => updateLang('spanish')}
      >
        SP
      </button>
      <button
        type='button'
        className={isEnglish ? 'btn__primary active' : 'btn__primary'}
        onClick={() => updateLang('english')}
      >
        ENG
      </button>
    </div>
  </div>
)

Options.propTypes = {
  isEnglish: PropTypes.bool.isRequired,
  updateLang: PropTypes.func.isRequired
}

export default Options
