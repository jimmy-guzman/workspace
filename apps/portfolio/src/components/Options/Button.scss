@mixin e($element) {
  &__#{$element} {
    @content;
  }
}

@function em($target, $context: 18px) {
  @return ($target/$context) * 1em;
}

@mixin button-setup($type) {
  @extend %btn;
  @extend %btn__#{$type};

  &:focus {
    @extend %btn__#{$type};
  }
  &:hover {
    @extend %btn__#{$type}--hover;
  }
}

%btn {
  background: transparent;
  border: none;
  outline: none;
  font-size: em(24px);
  font-family: 'Oswald', Impact, 'Franklin Gothic Bold', sans-serif;
  cursor: pointer;
  transition: all 0.3s ease-in-out;
}

%btn__primary {
  color: #d7d5d9;
  fill: #d7d5d9;
}

%btn__primary--hover {
  color: #292b2f;
  fill: #40af93;
}

%btn__secondary {
  color: #292b2f;
  fill: #292b2f;
}

%btn__secondary--hover {
  color: #292b2f;
  fill: #40af93;
}

.btn {
  @include e(primary) {
    @include button-setup(primary);
  }
  @include e(secondary) {
    @include button-setup(secondary);
  }
}
