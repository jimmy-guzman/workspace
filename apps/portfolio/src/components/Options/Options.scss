@function em($target, $context: 18px) {
  @return ($target/$context) * 1em;
}
@mixin e($element) {
  &__#{$element} {
    @content;
  }
}

.options {
  font-family: 'Oswald', Impact, 'Franklin Gothic Bold', sans-serif;
  background: #292b2f;
  width: 100%;
  display: flex;
  flex-flow: row;
  justify-content: space-between;
  align-items: center;
  @include e(day-night) {
    display: flex;
    flex-flow: row;
    justify-content: space-between;
    align-items: center;
  }
  button {
    &:hover,
    &.active {
      color: #40af93;
    }
  }
}

.home-link {
  padding: em(8px);
  background: #40af93;
  color: #292b2f;
  letter-spacing: 3px;
  max-width: 500px;
}
.active-home {
  background: inherit;
  color: inherit;
  padding: 0;
  max-width: 0;
  overflow: hidden;
}

a {
  text-decoration: none;
  fill: #d7d5d9;
  &:hover {
    color: #d7d5d9;
  }
  &.active {
    color: #40af93;
    fill: #40af93;
  }
}
