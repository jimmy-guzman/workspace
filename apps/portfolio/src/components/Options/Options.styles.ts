import styled from 'styled-components'

import { flex, em } from '@portfolio/utils'
import { Link } from '../Link/Link.styles'

const Options = styled.div`
  background: ${({ theme }) => theme.colors.tertiary};
  width: 100%;
  ${flex('flex', 'row', 'space-between', 'center')}
`
const HomeLink = styled(Link)`
  padding: ${em('8px')};
  background: ${({ theme }) => theme.colors.primary};
  color: ${({ theme }) => theme.colors.tertiary};
  letter-spacing: ${em('3px')};
  max-width: ${em('500px')};
  &.active-home {
    background: inherit;
    color: inherit;
    padding: 0;
    max-width: 0;
    overflow: hidden;
  }
`

export const Styled = { Options, HomeLink }
