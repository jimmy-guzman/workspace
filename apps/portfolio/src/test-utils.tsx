import React from 'react'
import { render, RenderOptions } from '@testing-library/react'
import { Router } from 'react-router-dom'
import { StyleSheetManager } from 'styled-components'
import { createMemoryHistory } from 'history'

import { ThemeProvider } from './providers'

const Providers: React.ComponentType<{ route?: string }> = ({
  children,
  route,
}: {
  route?: string
  children?: React.ReactNode
}) => {
  const history = createMemoryHistory()

  if (route) {
    history.push(route)
  }

  return (
    <StyleSheetManager disableVendorPrefixes>
      <Router history={history}>
        <ThemeProvider>{children}</ThemeProvider>
      </Router>
    </StyleSheetManager>
  )
}

Providers.displayName = 'ss'

const customRender = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
  route?: string
) => {
  const wrapper: React.ComponentType = ({
    children,
  }: {
    children?: React.ReactNode
  }) => <Providers route={route}>{children}</Providers>

  return render(ui, { wrapper, ...options })
}

export * from '@testing-library/react'

export { customRender as render }
