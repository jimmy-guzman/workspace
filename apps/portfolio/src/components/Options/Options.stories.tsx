import React from 'react'

import { Options } from './Options'
import { action } from '@storybook/addon-actions'
import { withMockRouter } from '../../utils'

export default {
  title: 'portfolio/components/Options',
  component: Options,
  decorators: [
    (storyFn: () => React.ReactNode) =>
      withMockRouter(storyFn(), window.location.pathname)
  ]
}

export const english = () => (
  <Options isEnglish updateLang={action('updateLang')} />
)
export const spanish = () => (
  <Options isEnglish={false} updateLang={action('updateLang')} />
)
