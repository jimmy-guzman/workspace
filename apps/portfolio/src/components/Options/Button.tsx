import React from 'react'

import './Button.scss'

interface ButtonProps {
  className?: string
  onClick: () => void
  children: React.ReactNode
}

export const Button = ({ className, onClick, children }: ButtonProps) => {
  return (
    <button className={className} type='button' onClick={onClick}>
      {children}
    </button>
  )
}
