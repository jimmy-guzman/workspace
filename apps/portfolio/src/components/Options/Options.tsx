import React from 'react'
import { NavLink } from 'react-router-dom'

import { Styled } from '../Button'

import './Options.scss'

export interface OptionsProps {
  isEnglish?: boolean
  updateLang: (lang: string) => void
  activeRoute?: string
}

export const Options = ({
  isEnglish,
  updateLang,
  activeRoute = '/',
}: OptionsProps) => (
  <div className='options'>
    <NavLink
      exact
      to={activeRoute}
      className='home-link'
      activeClassName='active-home'
    >
      Jimmy Guzman
    </NavLink>
    <div className='options__lang'>
      <Styled.Button
        primary
        active={!isEnglish}
        onClick={() => updateLang('spanish')}
      >
        SP
      </Styled.Button>
      <Styled.Button
        primary
        active={isEnglish}
        onClick={() => updateLang('english')}
      >
        ENG
      </Styled.Button>
    </div>
  </div>
)
