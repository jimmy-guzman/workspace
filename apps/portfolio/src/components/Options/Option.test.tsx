import React from 'react'
import { render, fireEvent } from '@portfolio/test-utils'

import { Options, OptionsProps } from './Options'

const mockUpdateLang = jest.fn()

const setupOptions = (props?: Pick<OptionsProps, 'isEnglish'>) => {
  return render(<Options {...props} updateLang={mockUpdateLang} />)
}

describe('<Options />', () => {
  it('should render default snapshot', () => {
    const { container } = setupOptions()

    expect(container.firstChild).toMatchSnapshot()
  })
  it('should call updateLang with correct language', () => {
    const { getByText } = setupOptions()

    fireEvent.click(getByText('SP'))

    expect(mockUpdateLang).toHaveBeenCalledWith('spanish')

    fireEvent.click(getByText('ENG'))

    expect(mockUpdateLang).toHaveBeenCalledWith('english')
  })
  it('should set correct class when spanish', () => {
    const { getByText } = setupOptions({
      isEnglish: false,
    })

    expect(getByText('ENG')).toHaveStyle('color: rgb(215, 213, 217)')
  })
  it('should set correct class when english', () => {
    const { getByText } = setupOptions({
      isEnglish: true,
    })

    expect(getByText('ENG')).toHaveStyle('color: rgb(64, 175, 147)')
  })
})
