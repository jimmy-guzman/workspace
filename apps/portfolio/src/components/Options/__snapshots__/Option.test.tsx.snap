// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Options /> should render default snapshot 1`] = `
<div
  class="options"
>
  <a
    aria-current="page"
    class="home-link active-home"
    href="/"
  >
    Jimmy Guzman
  </a>
  <div
    class="options__lang"
  >
    <button
      class="btn__primary active"
      type="button"
    >
      SP
    </button>
    <button
      class="btn__primary"
      type="button"
    >
      ENG
    </button>
  </div>
</div>
`;
